flyteadmin/pkg/errors/errors.go

```diff
@@ -109,8 +111,68 @@ func NewIncompatibleClusterError(ctx context.Context, errorMsg, curCluster strin
 	return statusErr
 }
 
-func NewWorkflowExistsDifferentStructureError(ctx context.Context, request *admin.WorkflowCreateRequest) FlyteAdminError {
-	errorMsg := "workflow with different structure already exists"
+func compareJsons(jsonArray1 jsondiff.Patch, jsonArray2 jsondiff.Patch) []string {
+	results := []string{}
+	map1 := make(map[string]jsondiff.Operation)
+	for _, obj := range jsonArray1 {
+		map1[obj.Path] = obj
+	}
+
+	for _, obj := range jsonArray2 {
+		if val, ok := map1[obj.Path]; ok {
+			result := fmt.Sprintf("%s: %s -> %s\t", obj.Path, obj.Value, val.Value)
+			results = append(results, result)
+		}
+	}
+	return results
+}
+
+func NewTaskExistsDifferentStructureError(ctx context.Context, request *admin.TaskCreateRequest, oldSpec *core.TaskTemplate, newSpec *core.TaskTemplate) FlyteAdminError {
+	errorMsg := "task with different structure already exists:\n"
+	// omit source code file object storage path
```

Comment:
args/2 is still used for the digest computation, right? any reason to omit it here?

Reply:
Yes, we still hash the whole task for digest.
The reason to omit file path shown in error message is that it already trapped in "different structure with same id" error, saying code file must be different. So it's trivial to hide the path changed information. cc @pingsutw

Reply:
Another concern of mine is what if the file path value (`"s3://my-s3-bucket/flytesnacks/..."`) being stored with different `key path` in the future, for example, `args/3`.